### PR TITLE
Quotes are optional

### PR DESCRIPTION
**Pull request by breuerfelix**
you can now add a note with `dn hello this is my awesome note` and just omit the quotes. you can now write down notes much faster.

also this is not a breaking change, quotes still work !

### CODE REVIEW

**Comment by tomlockwood** [comment]
This approach is easier for me to understand than the one in https://github.com/tomlockwood/dn/pull/2, but also does not give consistency across commands.  I'm not sure which approach is better yet. Your thoughts?

**Comment by breuerfelix** [comment]
now its consistent across all commands which take a message as an input :)) 

i think that this is a way easier approach and more readable :)

but this also has to be mentioned in the readme tho :D

**Comment by tomlockwood** [comment]
Really cool!  I'll give this a proper pull and review later today.  Thanks!

**Comment by djwgit** [comment]
@breuerfelix , a suggestion
could you add some logic in `dn`, `dno` to prevent writing an empty line into the note as well?
say
```$dn [enter]```
this will insert an empty line into the note... 
thanks...

File: dn.sh [hunk 1]
```diff
@@ -1,11 +1,11 @@
 
 # Daily Notes Aliases
 dn() {
-  echo " * $1" >> ~/dn/$(date "+%Y-%m-%d")
+  echo " * $*" >> ~/dn/$(date "+%Y-%m-%d")
 }
 
 dno() {
-  echo " * $2" >> ~/dn/$1
+  echo " * ${*:2}" >> ~/dn/$1
```
Comment (djwgit) [inline dn.sh:8]: this inserts the date into the line in zsh (did not try on bash tho), need a change...

Comment (djwgit) [reply dn.sh:8]: this works on zsh.
```echo " * ${@:2}" >> ~/dn/$1```

**Comment by tomlockwood** [comment]
I want to keep this open and discuss some more.

It may boil down to personal preference but I actually prefer quotes being enforced.  Shell escape character errors are annoying to me, and there are less if I quote the string.

On the other hand, I know my quote approach is still possible with this new code.

But, I want to make this tool simple, as well, so having two different input formats seems like adding complexity.

But, also, your solution is elegant and easy to understand, so when I read the code I know what is going on.

Anyway, would love to hear the thoughts of others.  Thanks!

**Comment by djwgit** [comment]
dn "this is my note"
dn this is my note
knowing dn is a command-line tool, the rest is the note content, 
this first one looks more nature to me... 
the 2nd one, make you think, each one is an argument for the command...

**Comment by breuerfelix** [comment]
i made a 'lazy git' command, where you just do `lg this is my commit message` which is really handy once you get used to it :)

well i could understand your point but since this refactor is non breaking (quoting also works) i wouldnt discuss at all :D but yeah just my opinion :)

**Comment by tomlockwood** [comment]
If I think about my goals for this project PR #2 and this pull request don't really align - I want the functionality to be very simple to understand.  Unfortunately, I don't think this code fits that goal.

**Pull request closed without merging**